From bfc732dc04a0f1873d68f1e5be06c341a8959259 Mon Sep 17 00:00:00 2001
From: Luminarys <eumen@bensonhillbio.com>
Date: Wed, 24 Jun 2015 16:02:45 -0500
Subject: [PATCH] updated annotation query

---
 php/annotation_query.php |   35 +++++++++++++++++++++++++++++++++++
 1 files changed, 35 insertions(+), 0 deletions(-)

diff --git a/php/annotation_query.php b/php/annotation_query.php
index 8626be3..fdfa710 100644
--- a/php/annotation_query.php
+++ b/php/annotation_query.php
@@ -7,6 +7,36 @@ error_reporting(E_ALL | E_STRICT);
 //Load the settings
 $settings = parse_ini_file("../settings.ini");	
 
+$dbInit = 'mysql:host=' . $settings["server"] . ';dbname=' . $settings["orthodb"] .';charset=utf8';
+$db = new PDO($dbInit, $settings["user"], $settings["password"]);
+$db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+$db->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+
+$query = $db->prepare("SELECT * FROM OrthoMapReference");
+$validSpecies = array();
+if($query->execute()){
+	$result = $query->fetchAll();
+	foreach ($result as $spec){
+		array_push($validSpecies, $spec["prefix"]);
+	}
+}else{
+	echo "Warning, no MapReference table defined, exiting";
+	exit();
+}
+//SELECT ortho_prefix FROM OrthoMapOut WHERE network_prefix = ;
+$query = $db->prepare("SELECT * FROM OrthoMapOut");
+$species = $_GET["spec"];
+$dataSet = array();
+if($query->execute()){
+	$result = $query->fetchAll();
+	foreach($result as $spec){
+		$dataSet[$spec['network_prefix']] = $spec['ortho_prefix'];
+	}
+}else{
+	echo "Warning, no MapReference table defined, exiting";
+	exit();
+}
+
 //Initialize Server, using settings
 $dbInit = 'mysql:host=' . $settings["server"] . ';dbname=' . $settings["maindb"] .';charset=utf8';
 $db = new PDO($dbInit, $settings["user"], $settings["password"]);
@@ -131,6 +161,11 @@ $dbInit = 'mysql:host=' . $settings["server"] . ';dbname=' . $settings["orthodb"
 $db = new PDO($dbInit, $settings["user"], $settings["password"]);
 $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
 $db->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+$alt_data = array();
+foreach ($alt_specs as $ortho){
+	array_push($alt_data, $dataSet[$ortho]);
+}
+$alt_specs = array_unique($alt_data);
 foreach ($alt_specs as $ortho){
 	$gene = $species;
 	$inp = array($gene."_Genes.name",$ortho."_Genes.name",$gene."_Genes",$gene."_Ortho",$gene."_Ortho.id",$gene."_Genes.id",$ortho."_Ortho",$ortho."_Ortho.orth_id",$gene."_Ortho.orth_id",$ortho."_Genes",$ortho."_Genes.id",$ortho."_Ortho.id",$gene."_Genes.name");
-- 
1.7.1

